src/Elasticsearch.Net/Connection/HttpConnection.cs

```diff
@@ -66,104 +64,88 @@ public virtual ElasticsearchResponse<Stream> PostSync(Uri uri, byte[] data, IReq
+			var r = this.CreateHttpRequest(uri, "PUT", data, requestSpecificConfig);
 			return this.DoAsyncRequest(r, data, requestSpecificConfig: requestSpecificConfig);
 		}
 
 		public virtual Task<ElasticsearchResponse<Stream>> Delete(Uri uri, byte[] data, IRequestConfiguration requestSpecificConfig = null)
 		{
-			var r = this.CreateHttpWebRequest(uri, "DELETE", data, requestSpecificConfig);
+			var r = this.CreateHttpRequest(uri, "DELETE", data, requestSpecificConfig);
 			return this.DoAsyncRequest(r, data, requestSpecificConfig: requestSpecificConfig);
 		}
-		
+
 		public virtual Task<ElasticsearchResponse<Stream>> Delete(Uri uri, IRequestConfiguration requestSpecificConfig = null)
 		{
-			var r = this.CreateHttpWebRequest(uri, "DELETE", null, requestSpecificConfig);
+			var r = this.CreateHttpRequest(uri, "DELETE", null, requestSpecificConfig);
 			return this.DoAsyncRequest(r, requestSpecificConfig: requestSpecificConfig);
 		}
 
-		private static void ThreadTimeoutCallback(object state, bool timedOut)
+		protected virtual HttpRequestMessage CreateHttpRequest(Uri uri, string method, byte[] data, IRequestConfiguration requestSpecificConfig)
 		{
-			if (timedOut)
-			{
-				HttpWebRequest request = state as HttpWebRequest;
-				if (request != null)
-				{
-					request.Abort();
-				}
-			}
+			var request = this.CreateHttpRequestMessage(uri, method, data, requestSpecificConfig);
+			this.SetBasicAuthenticationIfNeeded(uri, request, requestSpecificConfig);
+			return request;
 		}
 
-		protected virtual void AlterServicePoint(ServicePoint requestServicePoint)
+		private void EnableCompressionIfNeeded()
 		{
-			requestServicePoint.UseNagleAlgorithm = false;
-			requestServicePoint.Expect100Continue = false;
-			requestServicePoint.ConnectionLimit = 10000;
-			//looking at http://referencesource.microsoft.com/#System/net/System/Net/ServicePoint.cs
-			//this method only sets internal values and wont actually cause timers and such to be reset
-			//So it should be idempotent if called with the same parameters
-			if (this.ConnectionSettings.KeepAliveTime.HasValue && this.ConnectionSettings.KeepAliveInterval.HasValue)
-				requestServicePoint.SetTcpKeepAlive(true, this.ConnectionSettings.KeepAliveTime.Value, this.ConnectionSettings.KeepAliveInterval.Value);
-		}
+			if (!this.ConnectionSettings.EnableCompressedResponses &&
+			    !this.ConnectionSettings.EnableHttpCompression)
+				return;
 
-		protected virtual HttpWebRequest CreateHttpWebRequest(Uri uri, string method, byte[] data, IRequestConfiguration requestSpecificConfig)
-		{
-			var request = this.CreateWebRequest(uri, method, data, requestSpecificConfig);
-			this.SetBasicAuthenticationIfNeeded(uri, request, requestSpecificConfig);
-			this.SetProxyIfNeeded(request);
-			this.AlterServicePoint(request.ServicePoint);
-			return request;
+			_httpHandler.AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate;

```

Comment:
Pretty neat that this is enough to handle Decompression of the response stream, didn't expect that